python_modules/dagster/dagster/_core/definitions/definitions_class.py

```diff
@@ -142,6 +142,27 @@ def _attach_resources_to_jobs_and_instigator_jobs(
     schedules = schedules or []
     sensors = sensors or []
 
+    # Add jobs in schedules and sensors (dedupe if they appear in both)
+    jobs = [
+        *{
+            *jobs,
+            *[
+                schedule.job
+                for schedule in schedules
+                if isinstance(schedule, ScheduleDefinition)
+                and schedule.has_loadable_target()
+                and isinstance(schedule.job, (JobDefinition, UnresolvedAssetJobDefinition))
```

Comment:
this is needed to satisfy pyright because schedules and sensors can apparently have a GraphDefinition in the job field? Does that ever actually happen?